libraries/chain/smt_evaluator.cpp

```diff
@@ -81,9 +81,57 @@ void smt_set_setup_parameters_evaluator::do_apply( const smt_set_setup_parameter
    FC_ASSERT( _db.has_hardfork( STEEM_SMT_HARDFORK ), "SMT functionality not enabled until hardfork ${hf}", ("hf", STEEM_SMT_HARDFORK) );
 }
 
+struct smt_set_runtime_parameters_evaluator_visitor
+{
+   const smt_token_object& _token;
+   database& _db;
+
+   smt_set_runtime_parameters_evaluator_visitor( const smt_token_object& token, database& db ): _token( token ), _db( db ){}
+
+   typedef void result_type;
+
+   void operator()( const smt_param_windows_v1& param_windows ) const
+   {
+      _db.modify( _token, [&]( smt_token_object& token )
+      {
+         token.cashout_window_seconds = param_windows.cashout_window_seconds;
+         token.reverse_auction_window_seconds = param_windows.reverse_auction_window_seconds;
+      });
+   }
+
+   void operator()( const smt_param_vote_regeneration_period_seconds_v1& vote_regeneration ) const
+   {
+      _db.modify( _token, [&]( smt_token_object& token )
+      {
+         token.vote_regeneration_period_seconds = vote_regeneration.vote_regeneration_period_seconds;
+         token.votes_per_regeneration_period = vote_regeneration.votes_per_regeneration_period;
+      });
+   }
+
+   void operator()( const smt_param_rewards_v1& param_rewards ) const
+   {
+      _db.modify( _token, [&]( smt_token_object& token )
+      {
+         token.content_constant = param_rewards.content_constant;
+         token.percent_curation_rewards = param_rewards.percent_curation_rewards;
+         token.percent_content_rewards = param_rewards.percent_content_rewards;
+         token.author_reward_curve = param_rewards.author_reward_curve;
+         token.curation_reward_curve = param_rewards.curation_reward_curve;
+      });
+   }
+};
+
 void smt_set_runtime_parameters_evaluator::do_apply( const smt_set_runtime_parameters_operation& o )
 {
    FC_ASSERT( _db.has_hardfork( STEEM_SMT_HARDFORK ), "SMT functionality not enabled until hardfork ${hf}", ("hf", STEEM_SMT_HARDFORK) );
+
+   const auto* _token = _db.find< smt_token_object, by_control_account >( o.control_account );
+   FC_ASSERT( _token, "Any token does not exist for ${account}",("account", o.control_account) );
```

Comment:
I'm not sure what "any token does not exist" means. i realize this isn't user facing, but I feel like error messages should be self-documenting.